lib/assets/javascripts/twine.coffee

```diff
@@ -124,7 +125,9 @@
     elements[node.bindingId] ?= {}
 
   # Queues a refresh of the DOM, batching up calls for the current synchronous block.
-  Twine.refresh = ->
+  # The callback will be called once when the refresh has completed.
+  Twine.refresh = (callback) ->
+    refreshCallbacks.push(callback) if callback
     return if refreshQueued
     refreshQueued = true
     setTimeout(Twine.refreshImmediately, 0)
```

Comment:
`refreshImmediately` is synchronous so it's a bit of an antipattern to give it a callback

We need the global list of callbacks since a call to `Twine.refresh` will not execute `refreshImmediately` if there is a pending refresh already.